tests/unit/machine/gp/FITCInferenceMethod_unittest.cc

```diff
@@ -590,5 +655,428 @@ TEST(FITCInferenceMethod,get_marginal_likelihood_derivatives_sparse)
+
+	// build parameter dictionary
+	CMap<TParameter*, CSGObject*>* parameter_dictionary=new CMap<TParameter*, CSGObject*>();
+	inf->build_gradient_parameter_dictionary(parameter_dictionary);
+
+	// compute derivatives wrt parameters
+	CMap<TParameter*, SGVector<float64_t> >* gradient=
+		inf->get_negative_log_marginal_likelihood_derivatives(parameter_dictionary);
+
+	// get parameters to compute derivatives
+	TParameter* scale_param=inf->m_gradient_parameters->get_parameter("scale");
+	TParameter* sigma_param=lik->m_gradient_parameters->get_parameter("sigma");
+	TParameter* noise_param=inf->m_gradient_parameters->get_parameter("inducing_noise");
+	TParameter* weights_param=kernel->m_gradient_parameters->get_parameter("weights");
+		TParameter* mean_param=mean->m_gradient_parameters->get_parameter("mean");
+
+	float64_t dnlZ_sf2=scale*(gradient->get_element(scale_param))[0];
+	float64_t dnlZ_lik=(gradient->get_element(sigma_param))[0];
+	float64_t dnlZ_noise=(gradient->get_element(noise_param))[0];
+	float64_t dnlZ_mean=(gradient->get_element(mean_param))[0];
+
+	SGVector<float64_t> dnlz_weights_vec=gradient->get_element(weights_param);
+	SGMatrix<float64_t> dnlz_weights(dnlz_weights_vec.vector,t_dim, dim, false);
+	//diagonal elements are in the log domain
+	dnlz_weights(0,0)*=weight00;
+	dnlz_weights(1,1)*=weight11;
+
+	dnlZ_lik+=dnlZ_noise;
+
+	// comparison of partial derivatives of negative log marginal likelihood
+	// with result from GPML 3.5 package:
+	//
+	//mean: 0.080187488475807
+	//cov: 5.226913608720613
+	//lik: 0.095062798184398
+	//
+	//weights=[
+	// 0.014310121599512  -1.258335093159427   0.021881344855798
+	// 0                  0.004443216429718    -1.253522960589666]
+	//
+	//Note that in GPML 3.5, weights is a upper-triangular matrix
+	//diagonal elements of the matrix are in the log domain
+	//while off-diagonal elements are in natrual domain
+
+	EXPECT_NEAR(dnlz_weights(0,0), 0.014310121599512, 1E-10);
+	EXPECT_NEAR(dnlz_weights(0,1), -1.258335093159427, 1E-10);
+	EXPECT_NEAR(dnlz_weights(0,2), 0.021881344855798, 1E-10);
+
+	//dnlz_weights(1,0) should be different (non-zero)
+	//since the kernel matrix in the Shogun's implementation does not
+	//have the "upper triangular matrix" constraint
+	EXPECT_NEAR(dnlz_weights(1,1), 0.004443216429718, 1E-10);
+	EXPECT_NEAR(dnlz_weights(1,2), -1.253522960589666, 1E-10);
+
+
+	EXPECT_NEAR(dnlZ_lik, 0.095062798184398,1E-10);
+	EXPECT_NEAR(dnlZ_mean, 0.080187488475807, 1E-10);
+	EXPECT_NEAR(dnlZ_sf2, 5.226913608720613, 1E-10);
+
+	//Note that in the latest GPML3.5, derivatives wrt xu (covSEfact) does not support yet
+	//In Shohun's implementation, the derivatives wrt xu are supported. 
+	//TParameter* lat_param=inf->m_gradient_parameters->get_parameter("inducing_features");
+	//SGVector<float64_t> tmp=gradient->get_element(lat_param);
+	//SGMatrix<float64_t> deriv_lat(tmp.vector, dim, m, false);
+
+	// clean up
+	SG_UNREF(gradient);
+	SG_UNREF(parameter_dictionary);
+	SG_UNREF(inf);
```

Comment:
same here, the test is a bit too long.
I suggest that you code up some set_up method that initialises everything and then the unit tests are very short and seperate for the gradients?